src/utils/metrics.h

```diff
@@ -493,6 +518,26 @@ class counter : public metric
         return _adder.fetch_and_reset();
     }
 
+    // The snapshot collected has following json format:
+    // {
+    //     "name": "<metric_name>",
+    //     "value": ...
+    // }
+    // where "name" is the name of the counter in string type, and "value" is just current value
+    // of the counter fetched by `value()`, in integral type (namely int64_t).
+    void take_snapshot(json::JsonWriter &writer) override
+    {
+        writer.StartObject();
+
+        writer.Key("name");
+        json::json_encode(writer, prototype()->name().data());
+
+        writer.Key("value");
+        json::json_encode(writer, value());
+
+        writer.EndObject();
```

Comment:
it's same with `guage`,  why not put the `base class`?

Reply:
> it's same with `guage`, why not put the `base class`?

The reason is that `value()` is not an abstract method of the base class. For example, the percentile type does not need a `value()` method, since each has multiple values.